File: ios/Classes/Ads/LingYeInterstitialHandler.swift
import Foundation
import Flutter
import CJMobileAd

class LingYeInterstitialHandler: NSObject {
    
    
    static let shared: LingYeInterstitialHandler = LingYeInterstitialHandler()
    
    var initParams: [String: Any] = [:]
    var interstitialAd: CJInterstitialAd?

    func handleMethodCall(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        initParams = call.arguments as? [String: Any] ?? [:]
        if call.method == LingYeAdSdkMethodNames.loadAndShowInterstitialAd {
            loadAndShowInterstitialAd()
            result(true)
        }
        result(nil)
    }

    func loadAndShowInterstitialAd() {
        interstitialAd = CJInterstitialAd(resourceId: initParams["adId"] as? String ?? "")
        interstitialAd?.delegate = self
        interstitialAd?.loadData()
    }
    
}


extension LingYeInterstitialHandler: CJInterstitialAdDelegate {
    
    func interstitialAdDidLoad(_ interstitialAd: CJInterstitialAd, resourceId: String) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.interstitialAdLoadSuccess, arguments: ["code": 0, "message": "插屏加载成功"])
        interstitialAd.show(fromRootViewController: LingYeTools.shared.currentViewController() ?? UIViewController());
    }
    
    func interstitialAdLoadFailed(_ interstitialAd: CJInterstitialAd, error: any Error) {
        let msg = error.localizedDescription
        let code = -1
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.interstitialAdLoadFailed, arguments: ["code": code, "message": msg])
    }
    
    func interstitialAd(onShow interstitialAd: CJInterstitialAd) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.interstitialAdOnShow, arguments: ["code": 0, "message": "插屏展示成功"])
    }
    
    func interstitialAd(onClicked interstitialAd: CJInterstitialAd) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.interstitialAdOnClick, arguments: ["code": 0, "message": "插屏触发点击"])
    }
    
    func interstitialAd(onClosed interstitialAd: CJInterstitialAd) {
        LingYeBaseRegister.shared.sendToFlutter(method: LingYeAdSdkCallBackMethodNames.interstitialAdOnClose, arguments: ["code": 0, "message": "插屏触发关闭"])
    }
    
    func interstitialAdChannel(onClicked interstitialAd: CJInterstitialAd, extraParams params: [AnyHashable : Any]) {}
    
    func interstitialAdChannel(onShow interstitialAd: CJInterstitialAd, extraParams params: [AnyHashable : Any]) {}
    
    
}

File: android/src/main/kotlin/com/example/flutter_cjadsdk_plugin/ads/LingYeInterstitialHandler.kt
import android.content.Context
import android.graphics.Point
import android.util.DisplayMetrics
import android.view.ViewGroup
import android.view.WindowManager
import cj.mobile.CJInterstitial
import cj.mobile.CJSplash
import cj.mobile.listener.CJInterstitialListener
import cj.mobile.listener.CJSplashListener
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class LingYeInterstitialHandler {

    var interstitialAd: CJInterstitial? = null
    var params: Map<String, Any>? = null

    companion object {
        @Volatile
        private var instance: LingYeInterstitialHandler? = null
        fun getInstance(): LingYeInterstitialHandler {
            return instance ?: synchronized(this) {
                instance ?: LingYeInterstitialHandler().also { instance = it }
            }
        }
    }

    fun handleMethodCall(call: MethodCall, result: MethodChannel.Result) {
        val method: String = call.method
        val flutterParams: Map<String, Any>? = call.arguments as? Map<String, Any>
        params = flutterParams
        when (method) {
            LingYeAdSdkMethodNames.loadAndShowInterstitialAd -> {
                loadAndShowInterstitialAd()
                result.success(true)
            }
            else -> result.notImplemented()
        }
    }

    private fun loadAndShowInterstitialAd() {
        interstitialAd = CJInterstitial()
        val activity = LingYeTools.getActivity()
        val adId = params?.get("adId")
        val windowManager: WindowManager? = activity?.getSystemService(Context.WINDOW_SERVICE) as? WindowManager
        val display = windowManager?.defaultDisplay
        val size = Point()
        display?.getRealSize(size)
        val map: MutableMap<String, Any> = HashMap()
        interstitialAd?.loadAd(activity, adId.toString(), object : CJInterstitialListener {
            override fun onShow() {
                map.put("code", 0)
                map.put("message", "插屏展示成功")
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.interstitialAdOnShow,
                    map
                )
            }

            override fun onError(s: String, s1: String) {
                map["code"] = -1
                map["message"] = s+s1
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.splashAdLoadFailed,
                    map
                )
            }

            override fun onLoad() {
                map.put("code", 0)
                map.put("message", "插屏加载成功")
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.interstitialAdLoadSuccess,
                    map
                )
                interstitialAd?.showAd(activity)
            }

            override fun onClick() {
                map.put("code", 0)
                map.put("message", "插屏触发点击")
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.interstitialAdOnClick,
                    map
                )
            }

            override fun onClose() {
                map.put("code", 0)
                map.put("message", "插屏触发关闭")
                LingYeBaseRegister.getInstance().sendMessageToFlutter(
                    LingYeAdSdkCallBackMethodNames.interstitialAdOnClose,
                    map
                )
            }
        })
    }

}